modules/svg/processor.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package svg
+
+import (
+	"bytes"
+	"fmt"
+	"regexp"
+	"sync"
+)
+
+type normalizeVarsStruct struct {
+	reXMLDoc,
+	reComment,
+	reAttrXMLNs,
+	reAttrSize,
+	reAttrClassPrefix *regexp.Regexp
+}
+
+var (
+	normalizeVars     *normalizeVarsStruct
+	normalizeVarsOnce sync.Once
+)
+
+// Normalize normalizes the SVG content: set default width/height, remove unnecessary tags/attributes
+// It's designed to work with valid SVG content. For invalid SVG content, the returned content is not guaranteed.
+func Normalize(data []byte, size int) []byte {
```

Comment:
Didn't I review this function recently already?

Reply:
Yup, it comes from #26236 , I extracted the code because I think at least it's better than before.